src/Features/Core/Portable/IncrementalCaches/SymbolTreeInfoIncrementalAnalyzerProvider.cs

```diff
@@ -230,14 +230,21 @@ private async Task UpdateSourceSymbolTreeInfoAsync(Project project, Cancellation
                 }
             }
 
-            private Task GetTask(Project project, Func<Task> func, CancellationToken cancellationToken)
+            [PerformanceSensitive("https://github.com/dotnet/roslyn/issues/36158", AllowCaptures = false, Constraint = "Avoid captures to reduce GC pressure when running in the host workspace.")]
```

Comment:
Shouldn't the attribute be applied to `UpdateSymbolTreeInfoAsync` above?

Reply:
➡️ No. `UpdateSymbolTreeInfoAsync` was not a significant source of allocations in the test scenario. The changes to the method are a side effect of the signature change, as opposed to a change directly intended to reduce allocations.